spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/PublisherCallbackChannelImpl.java

```diff
@@ -872,6 +882,10 @@ public void addListener(Listener listener) {
 				if (pendingConfirm.getTimestamp() < cutoffTime) {
 					expired.add(pendingConfirm);
 					iterator.remove();
+					CorrelationData correlationData = pendingConfirm.getCorrelationData();
+					if (correlationData != null && StringUtils.hasText(correlationData.getReturnCorrelation())) {
+						this.pendingReturns.remove(correlationData.getReturnCorrelation());
```

Comment:
Why an existing `CorrelationData.id` can't be used for this return purpose as well?

Reply:
Good idea!